src/pages/workspaces/workspace/analysis/utils/cost-utils.ts

```diff
@@ -213,74 +259,97 @@ export const getAzureDiskCostEstimate = ({ region, persistentDiskSize }) => {
+  if (isAzureConfig(runtimeConfig)) {
     return Utils.switchCase(
       status,
-      ['Stopped', () => 0.0],
-      ['Error', () => 0.0],
+      [runtimeStatuses.stopped.leoLabel, () => 0.0],
+      [runtimeStatuses.error.leoLabel, () => 0.0],
       [Utils.DEFAULT, () => getAzureComputeCostEstimate(runtimeConfig)]
     );
   }
-  return Utils.switchCase(
-    status,
-    ['Stopped', () => runtimeConfigBaseCost(runtimeConfig)],
-    ['Error', () => 0.0],
-    [Utils.DEFAULT, () => runtimeConfigCost(runtimeConfig)]
-  );
+  if (isGceRuntimeConfig(runtimeConfig) || isDataprocConfig(runtimeConfig)) {
+    return Utils.switchCase(
+      status,
+      [runtimeStatuses.stopped.leoLabel, () => runtimeConfigBaseCost(runtimeConfig)],
+      [runtimeStatuses.error.leoLabel, () => 0.0],
+      [Utils.DEFAULT, () => runtimeConfigCost(runtimeConfig)]
+    );
+  }
+  throw new Error(`Unknown runtime config type ${runtimeConfig}`);
 };
 
 export const getCostForDisk = (
-  app,
-  appDataDisks,
-  computeRegion,
-  currentRuntimeTool,
-  persistentDisks,
-  runtimes,
-  toolLabel
-) => {
-  let diskCost = '';
-  const curPd = persistentDisks && persistentDisks.length && getCurrentPersistentDisk(runtimes, persistentDisks);
+  app: App | undefined,
+  appDataDisks: PersistentDisk[],
+  computeRegion: string,
+  currentRuntimeTool: RuntimeToolLabel,
+  persistentDisks: PersistentDisk[],
+  runtimes: Runtime[],
+  toolLabel: ToolLabel
+): number => {
+  let diskCost = 0;
+  const rawPd = persistentDisks && persistentDisks.length && getCurrentPersistentDisk(runtimes, persistentDisks);
+  const curPd = rawPd && rawPd.diskType && updatePdType(rawPd);
   if (curPd && isAzureDisk(curPd)) {
-    return getAzureDiskCostEstimate({ region: computeRegion, persistentDiskSize: curPd.size }) / numberOfHoursPerMonth;
+    return getAzureDiskCostEstimate(computeRegion, curPd.size) / numberOfHoursPerMonth;
   }
   if (currentRuntimeTool === toolLabel && persistentDisks && persistentDisks.length) {
-    const { size = 0, status = 'Running', diskType = pdTypes.standard } = curPd || {};
+    const { size = 0, status = diskStatuses.ready.leoLabel, diskType = pdTypes.standard } = curPd || {};
     diskCost = getPersistentDiskCostHourly({ size, status, diskType }, computeRegion);
   } else if (app && appDataDisks && toolLabel === appToolLabels.GALAXY) {
     const currentDataDisk = getCurrentAttachedDataDisk(app, appDataDisks);
     // Occasionally currentDataDisk will be undefined on initial render.
-    diskCost = currentDataDisk ? getGalaxyDiskCost(currentDataDisk) : '';
+    diskCost = currentDataDisk ? getGalaxyDiskCost(currentDataDisk) : 0;
   }
   return diskCost;
 };
 
-export const getCostDisplayForTool = (app, currentRuntime, currentRuntimeTool, toolLabel) => {
+export const getCostDisplayForTool = (
+  app: App | undefined,
+  currentRuntime: Runtime | undefined,
+  currentRuntimeTool: RuntimeToolLabel | undefined,
+  toolLabel: ToolLabel | undefined
```

Comment:
In one of the tests, currentRuntime is of type ToolLabel, but the variable isn't appended with "Label."

Could we update the variable "currentRuntimeTool" to be "currentRuntimeToolLabel" since it's just the string, not a "Tool" object?

<img width="612" alt="Screenshot 2023-05-08 at 2 58 55 PM" src="https://user-images.githubusercontent.com/11773357/236909188-50b1291b-3e86-4bc0-8d9b-2ad3e2ab2889.png">